src/main/java/Transcription/CacheKey.java

```diff
@@ -1,21 +1,18 @@
-package Search;
-
-import SpeechToText.SpeechToText;
+package Transcription;
 
 import java.io.File;
 
 public class CacheKey {
-
     private static final String CACHE_DIRECTORY = ".cache";
     private static final String TRANSCRIPT_FILE_EXTENSION = ".transcript";
     private static final String TIMESTAMP_FILE_EXTENSION = ".timestamp";
 
     public String filename;
     public String lastModified;
-    public SpeechToText speechToText;
+    public VoskAdapter speechToText;
```

Comment:
Also recommend making this an enum instead of passing the whole speechToText engine to the cache key.

Reply:
I think you mentioned previously that this could be solved with dependency injection